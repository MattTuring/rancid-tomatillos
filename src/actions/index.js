export const addMovies = movies => ({
    type: 'ADD_MOVIES',
    movies
});

export const addRatings = ratings => ({
    type: 'ADD_RATING',
    ratings
});

export const addUserState = userInfo => ({
  type: 'ADD_USER_STATE',
  userInfo
});

export const updateLoggedIn = userInfo => ({
  type: 'UPDATE_IS_LOGGEDIN',
  userInfo
});
