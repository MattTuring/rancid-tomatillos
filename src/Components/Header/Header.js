import React from 'react'
import {Link} from 'react-router-dom'
import filmStrip from '../../Images/film-strip.png'
import { connect } from 'react-redux'
import { updateLoggedIn } from '../../actions';


const Header = (login) => {
  console.log(login)
  return (
    <>
    <div className='header-background'>
      <nav>
        <h1>RANCID TOMATILLOS</h1>
        {login.login.loggedIn ? <Link to='/'>
        <button className='login-button' onClick={() => { login.updateLoggedIn({
          id:'',
          username: '',
          password: '',
          error: null,
          loggedIn: false
        }) }}>LOG&nbsp;OUT</button>
        </Link> : <Link to='/login'>
          <button className='login-button'>LOG&nbsp;IN</button>
        </Link>}
      </nav>
    </div>

    <div className='film-strip' style={{backgroundImage: `url(${filmStrip})`}}>
    </div>
    </>
  )
}

const mapStateToProps = state => ({
  login: state.login
})
const mapDispatchToProps = dispatch => ({
  updateLoggedIn: login => dispatch(updateLoggedIn(login))
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)
