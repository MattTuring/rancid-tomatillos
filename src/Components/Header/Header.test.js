import { addUserState, addRatings, changeLoading, updateLoggedIn } from '../../actions';
import { mapDispatchToProps, matchStateProps, Header } from './Header'
import { shallow } from 'enzyme';
import React from 'react';

describe('Header', () => {
  let wrapper;
  let mockUpdateLoggedIn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Header
      user={{user: "Lucy"}}
      updateLoggedIn={mockUpdateLoggedIn}
    />)
  })

  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('calls dispatch with an addRatings action when addUserState is called', () =>{
    const mockDispatch = jest.fn();
    const actionToDispatch = updateLoggedIn([{
      id:'',
      username: '',
      ratings: [],
      loggedIn: false
      }]);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.updateLoggedIn([{
      id:'',
      username: '',
      ratings: [],
      loggedIn: false
    }]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
