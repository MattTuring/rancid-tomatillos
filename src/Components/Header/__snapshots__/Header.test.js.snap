// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<Fragment>
  <div
    className="header-background"
  >
    <nav>
      <h1>
        RANCID TOMATILLOS
      </h1>
      <Link
        to="/login"
      >
        <button
          className="login-button"
        >
          LOGÂ IN
        </button>
      </Link>
    </nav>
  </div>
  <div
    className="film-strip"
    style={
      Object {
        "backgroundImage": "url(film-strip.png)",
      }
    }
  />
</Fragment>
`;
